server/tests/src/test/java/org/infinispan/server/persistence/TableManipulation.java

```diff
@@ -0,0 +1,81 @@
+package org.infinispan.server.persistence;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import org.infinispan.commons.marshall.ProtoStreamMarshaller;
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.persistence.jdbc.configuration.JdbcStringBasedStoreConfigurationBuilder;
+import org.infinispan.persistence.jdbc.configuration.PooledConnectionFactoryConfiguration;
+import org.infinispan.persistence.jdbc.configuration.PooledConnectionFactoryConfigurationBuilder;
+import org.infinispan.persistence.jdbc.connectionfactory.ConnectionFactory;
+import org.infinispan.persistence.jdbc.impl.connectionfactory.PooledConnectionFactory;
+import org.infinispan.persistence.jdbc.impl.table.TableManager;
+import org.infinispan.persistence.jdbc.impl.table.TableManagerFactory;
+import org.infinispan.persistence.keymappers.DefaultTwoWayKey2StringMapper;
+
+public class TableManipulation {
+
+   private String cacheName;
+   private ConnectionFactory connectionFactory;
+   private Connection connection;
+   private PooledConnectionFactoryConfigurationBuilder persistenceConfiguration;
+   private ConfigurationBuilder configurationBuilder;
+
+   public TableManipulation(String cacheName, PooledConnectionFactoryConfigurationBuilder persistenceConfiguration, ConfigurationBuilder configurationBuilder) {
+      this.cacheName = cacheName;
+      this.persistenceConfiguration = persistenceConfiguration;
+      this.configurationBuilder = configurationBuilder;
+   }
+
+   private ConnectionFactory getConnectionFactory() {
```

Comment:
You need to make sure the resources returned by `TableManipulation` are always closed after a test has finished ... this is probably the cause of the `ThreadLeakChecker` failures.

Reply:
@ryanemerson I fixed some `ThreadLeakChecker` but there's one ThreadLeak that is still happening, I don't know why it's happening.

Reply:
@wburns Can you remember what was leaking the `agroal` connection on your PR before?

Reply:
Mine was a bug in the start method of the PersistenceManagerImpl, I highly doubt it is related and it shouldn't be able to resurface now.

https://github.com/infinispan/infinispan/blob/master/core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java#L198

Reply:
@ryanemerson I removed the dependency from `TableManager` and fixed all thread leaker